File: Examples/test-suite/ruby/li_typemaps_runme.rb
#!/usr/bin/env ruby
#
# Tests the typemaps.i INPUT/OUTPUT/INOUT typemaps

require 'swig_assert'
require 'li_typemaps'

include Li_typemaps

swig_assert_equal_simple(Li_typemaps.in_int(7), 7)
swig_assert_equal_simple(Li_typemaps.inr_int(7), 7)
swig_assert_equal_simple(Li_typemaps.out_int(7), 7)
swig_assert_equal_simple(Li_typemaps.inout_int(7), 7)
swig_assert_equal_simple(Li_typemaps.inoutr_int(7), 7)

# Two parameters using the same typemap
swig_assert_equal_simple(Li_typemaps.in_int_multi(3, 4), 7)
swig_assert_equal_simple(Li_typemaps.out_int_multi(5, 6), [5, 6])
swig_assert_equal_simple(Li_typemaps.inout_int_multi(7, 8), [14, 24])

# A reference cannot be bound to a null pointer, so nil is rejected rather than dereferenced in the call
def check_nullref_rejected(name)
  begin
    yield
  rescue StandardError
    return
  end
  raise "#{name} accepted nil for a reference"
end

check_nullref_rejected('inr_int') { Li_typemaps.inr_int(nil) }
check_nullref_rejected('inoutr_int') { Li_typemaps.inoutr_int(nil) }

File: Examples/test-suite/python/li_typemaps_runme.py
from li_typemaps import *
from swig_test_utils import swig_assert_raises, swig_check

swig_check(in_int(7), 7)
swig_check(inr_int(7), 7)
swig_check(out_int(7), 7)
swig_check(inout_int(7), 7)
swig_check(inoutr_int(7), 7)

# Two parameters using the same typemap
swig_check(in_int_multi(3, 4), 7)
swig_check(out_int_multi(5, 6), [5, 6])
swig_check(inout_int_multi(7, 8), [14, 24])

# A reference cannot be bound to a null pointer, so None is rejected rather than dereferenced in the call
with swig_assert_raises(TypeError):
    inr_int(None)
with swig_assert_raises(TypeError):
    inoutr_int(None)
